x/ccv/provider/keeper/proposal.go

```diff
@@ -342,91 +349,71 @@ func (k Keeper) BeginBlockInit(ctx sdk.Context) {
 	k.DeletePendingConsumerAdditionProps(ctx, propsToExecute...)
 }
 
-// ConsumerAdditionPropsToExecute iterates over the pending consumer addition proposals
-// and returns an ordered list of proposals to be executed, ie. consumer clients to be created.
+// GetConsumerAdditionPropsToExecute returns the pending consumer addition proposals
+// that are ready to be executed, i.e., consumer clients to be created.
 // A prop is included in the returned list if its proposed spawn time has passed.
 //
 // Note: this method is split out from BeginBlockInit to be easily unit tested.
-func (k Keeper) ConsumerAdditionPropsToExecute(ctx sdk.Context) []types.ConsumerAdditionProposal {
-
-	// store the (to be) executed proposals in order
+func (k Keeper) GetConsumerAdditionPropsToExecute(ctx sdk.Context) []types.ConsumerAdditionProposal {
 	propsToExecute := []types.ConsumerAdditionProposal{}
-
-	iterator := k.PendingConsumerAdditionPropIterator(ctx)
-	defer iterator.Close()
-
-	k.IteratePendingConsumerAdditionProps(ctx, func(spawnTime time.Time, prop types.ConsumerAdditionProposal) (stop bool) {
-		if !ctx.BlockTime().Before(spawnTime) {
+	k.IteratePendingConsumerAdditionProps(ctx, func(prop types.ConsumerAdditionProposal) (stop bool) {
+		if !ctx.BlockTime().Before(prop.SpawnTime) {
 			propsToExecute = append(propsToExecute, prop)
 			return false // do not stop the iteration
 		}
-		return true // stop iteration, proposals are ordered by spawn time, so no additional pending props are ready to act upon
+		// stop iteration; proposals are ordered by spawn time,
+		// so no additional pending props are ready to act upon
+		return true
 	})
-
 	return propsToExecute
 }
 
+// IteratePendingConsumerAdditionProps iterates through the pending consumer addition proposals.
+//
+// Note that the pending consumer addition proposals are stored under keys with the following format:
+// PendingCAPBytePrefix | spawnTime | chainID
+// Thus, the iteration is in ascending order of spawnTimes.
 func (k Keeper) IteratePendingConsumerAdditionProps(
 	ctx sdk.Context,
-	cb func(spawnTime time.Time, prop types.ConsumerAdditionProposal) (stop bool),
+	cb func(prop types.ConsumerAdditionProposal) (stop bool),
 ) {
-	iterator := k.PendingConsumerAdditionPropIterator(ctx)
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{types.PendingCAPBytePrefix})
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		key := iterator.Key()
-		spawnTime, _, err := types.ParsePendingCAPKey(key)
-		if err != nil {
-			panic(fmt.Errorf("failed to parse pending client key: %w", err))
-		}
-
 		var prop types.ConsumerAdditionProposal
 		k.cdc.MustUnmarshal(iterator.Value(), &prop)
 
-		stop := cb(spawnTime, prop)
+		stop := cb(prop)
 		if stop {
 			break
 		}
```

Comment:
You could instead the name the callback `stop` and have

```
if stop(prop) {
  return
}
```

Reply:
Is there an advantage to do so besides saving 3 lines of code. The `cb` name is more accurate since it's a callback. Given that we had problems with this kind of callbacks, I prefer this approach.